packages/service-library/src/servicelib/rabbitmq.py

```diff
@@ -211,13 +223,21 @@ async def rpc_request(
             namespace, method_name
         )
         try:
-            queue_expiration_timeout = timeout_s
-            awaitable = self._rpc.call(
-                namespaced_method_name,
-                expiration=queue_expiration_timeout,
-                kwargs=kwargs,
-            )
-            return await asyncio.wait_for(awaitable, timeout=timeout_s)
+            async for attempt in AsyncRetrying(
+                wait=wait_random(2),
+                stop=stop_after_delay(connection_error_timeout_s),
+                retry=retry_if_exception(AMQPConnectionError),
+                before=before_log(log, logging.DEBUG),
+                reraise=True,
+            ):
+                with attempt:
+                    queue_expiration_timeout = timeout_s
+                    awaitable = self._rpc.call(
```

Comment:
Q: does this happen a lot? Why did you decide to do a retry on every single rpc call. Sometimes retry are not the best option since they can saturate the destination (see circuit breaker pattern)